scripts/cluster/setup_master_node.go

```diff
@@ -199,18 +215,42 @@ func InstallKnativeServingComponent(stockContainerd string) error {
 		if !utils.CheckErrorWithMsg(err, "Failed to install Knative Serving component!\n") {
 			return err
 		}
-		_, err = utils.ExecShellCmd("kubectl apply -f https://github.com/knative/serving/releases/download/knative-v%s/serving-core.yaml", configs.Knative.KnativeVersion)
-		if !utils.CheckErrorWithTagAndMsg(err, "Failed to install Knative Serving component!\n") {
-			return err
+		
+		if _, fileStat := os.Stat(path.Join(configs.VHive.VHiveRepoPath, path.Join("configs/knative_yamls", "container-serving-core.yaml"))); fileStat != nil {
```

Comment:
Didn't follow the logic: if we have valid fileStat we install from the Internet, right? Sounds wrong, should be the opposite.